### PR TITLE
Feature/notifications feed

### PR DESCRIPTION
**Pull request by Zoeshwa**

### CODE REVIEW

File: apps/ui/src/utils/format/formatDateRelative.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+const formatDateRelative = (date: Date): string => {
```
Comment (Kovler10) [inline apps/ui/src/utils/format/formatDateRelative.ts:1]: try looking at day.js they should have functions that do that already

File: apps/ui/src/utils/messages/actionTextDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import { Message, MessageType } from '@communecar/types';
+
+const actionTextDisplay = (message: Message) => {
```
Comment (Kovler10) [inline apps/ui/src/utils/messages/actionTextDisplay.tsx:3]: is this a component?

File: apps/ui/src/utils/messages/actionTextDisplay.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const entityNameTextStyle = <strong>{` "${message.entityName}" `}</strong>;
```
Comment (Kovler10) [inline apps/ui/src/utils/messages/actionTextDisplay.tsx:4]: use CSS instead of strong

File: packages/types/src/Messages.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { User } from './User';
+
+export type Message = {
```
Comment (Kovler10) [inline packages/types/src/Messages.ts:3]: move exports to end of file

File: apps/ui/src/Pages/Messages/MessageCard.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+}) => {
+  const { submitRespondToMessage, isLoading, error } = useRespondToMessage();
+
+  const handleAction = async (action: RequestActions) => {
+    const success = await submitRespondToMessage(message.id, action);
+    if (success && !isLoading) {
+      alert(`Action ${action} was successful!`);
+      onActionComplete(message.id, true);
+    } else {
+      alert(`Failed to ${action} the request: ${error?.message}`);
```
Comment (Kovler10) [inline apps/ui/src/Pages/Messages/MessageCard.tsx:34]: change this to use toasts or snackbars

File: apps/ui/src/Pages/Messages/MessageCard.tsx [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  };
+
+  const isRequestType = [
+    MessageType.JOINING_COMMUNITY_REQUEST,
+    MessageType.JOINING_RIDE_REQUEST,
+  ].includes(message.type);
+
+  const messageText = (
+    <>
+      <strong>{`${message.creatorUser.firstName} `}</strong>
```
Comment (Kovler10) [inline apps/ui/src/Pages/Messages/MessageCard.tsx:45]: use CSS instead of strong tag

Comment (Kovler10) [reply apps/ui/src/Pages/Messages/MessageCard.tsx:45]: there's no need for string interpolation here

File: apps/ui/src/Pages/Messages/MessageCard.tsx [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+  return (
+    <ListItem
+      divider
+      sx={{ alignItems: 'flex-start', flexDirection: 'column' }}
+    >
+      <Box sx={{ display: 'flex', flexDirection: 'row', alignItems: 'center' }}>
+        <ListItemAvatar>
+          <Avatar
+            src={message.creatorUser.avatarUrl || undefined}
```
Comment (Kovler10) [inline apps/ui/src/Pages/Messages/MessageCard.tsx:58]: use optional chaining

File: apps/ui/src/hooks/Messages/useMessagesForUser.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect, useCallback } from 'react';
+import { fetchMessagesForUser } from '../../apis/messages/fetchMessagesForUser';
+import { Message } from '@communecar/types';
+
+const useMessagesForUser = (userId: string) => {
```
Comment (Kovler10) [inline apps/ui/src/hooks/Messages/useMessagesForUser.ts:5]: useUserMessages

File: apps/ui/src/Pages/Messages/MessageCard.tsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      </Box>
+      {isRequestType && (
+        <Box
+          sx={{
+            display: 'flex',
+            flexDirection: 'row',
+            bottom: 0,
+          }}
+        >
+          <Button
```
Comment (Kovler10) [inline apps/ui/src/Pages/Messages/MessageCard.tsx:80]: how about icon buttons instead? to be more simple

**Review by Kovler10 — CHANGES_REQUESTED** [review]
Nice! left some comments

File: apps/ui/src/ProtectedRoute.tsx [hunk 10]
```diff
@@ -4,6 +4,8 @@ import { useUser } from './hooks/Users/useUser';
 const ProtectedRoute = () => {
   const { user } = useUser();
 
+  return <Outlet />;
+
   return !user ? <Navigate to="/" replace /> : <Outlet />;
```
[the remark this answers is not in the document]
Comment (Zoeshwa) [reply apps/ui/src/ProtectedRoute.tsx:9]: yes you right i did this for myself and forget to change back

File: apps/ui/src/Pages/Messages/MessageCard.tsx (hunk 5 above)
Comment (Zoeshwa) [reply apps/ui/src/Pages/Messages/MessageCard.tsx:34]: we talk about this - we will do this to the whole app in a diffrent branch

File: apps/ui/src/Pages/Messages/MessageCard.tsx (hunk 6 above)
Comment (Zoeshwa) [reply apps/ui/src/Pages/Messages/MessageCard.tsx:45]: This is for the white space

**Review by Kovler10 — APPROVED** [review]

**Pull request merged**